mslib/utils/auth.py

```diff
@@ -100,14 +98,21 @@ def get_auth_from_url_and_name(server_url, http_auth, overwrite_login_cache=True
         if server_url == url:
             try:
                 password = get_password_from_keyring(service_name=url, username=auth_name)
+                if password is None:
+                    password = ""  # Ensure password is a string
             except keyring.errors.NoKeyringError as ex:
-                password = None
+                password = ""
                 logging.info("Can't use Keyring on your system: %s" % ex)
-            if overwrite_login_cache and password is not None and password.strip() != "":
+
+            if overwrite_login_cache and password.strip() != "":
                 constants.AUTH_LOGIN_CACHE[server_url] = (auth_name, password)
+
             name = auth_name
             break
+
     if name == "":
         name = None
-    auth = constants.AUTH_LOGIN_CACHE.get(server_url, (name, None))
+
+    # Always return a valid string for the password
+    auth = constants.AUTH_LOGIN_CACHE.get(server_url, (name, ""))
```

Comment:
We don't accept empty passwords. And should not start using them in the AUTH_LOGIN_CACHE.